src/dstv_element.rs

```diff
@@ -46,4 +53,5 @@ pub trait DstvElement {
     /// Returns the flange code of the element.
     /// This is used to determine the side of the element that is drawn.
     fn get_facing(&self) -> &PartFace;
+    fn as_any(&self) -> &dyn core::any::Any;
```

Comment:
Thanks for the reply!  I understand being busy and not noticing things.

I added the `as_any`, because otherwise, I don't know a way to cast the generic trait as another type, as they are all stored in the same Vec (and present the same type).

```rust
for elem in &value.elements {
  let elem = elem.as_any();
  if let Some(hole) = elem.downcast_ref::<DstvHole>() {
      holes.push(hole.into());
  } else if let Some(border) = elem.downcast_ref::<OuterBorder>() {
      borders.0.push(border);
  // etc
  }
}
```

Reply:
Ah yes, I see. I believe we can have a better approach by introducing Enum for types:
```rust 
pub struct Dstv {
    /// The header of the DSTV file containing information about the project, phase, etc.
    pub header: Header,
    /// A vector of DSTV elements, e.g., cuts, holes, borders, etc.
    pub elements: Vec<DstvElementType>,
}

pub enum DstvElementType {
    Hole(Hole),
    OuterBorder(OuterBorder),
    // etc...
}
```
It would take more effort, but in that case, it would become more efficient to check for the element type to perform the logic you want:
```rust
for elem in &dstv.elements {
    match elem {
        DstvElementType::Hole(hole) => {
            holes.push(hole.clone());
        }
        DstvElementType::OuterBorder(outerborder) => {
            borders.0.push(outerborder);
        }
    }
}
```
Then, we eliminate the `dyn` dynamic dispatching, the 'consuming' code becomes more readable, and no more downcasting is required. 

I'm more than happy to work this out in the upstream repository if this works for you as well.

Reply:
Oh, yes, enums are great.  I agree, this would be the way to go.

Reply:
In that case, and for convenience, let's merge this PR upstream. For the time being, you can use the `as_any` feature. Once I've implemented the enumeration implementation, There are a lot of helpful changes in this PR I can leverage while making the change.

Reply:
https://github.com/tluijken/dstv/pull/3 has been merged into the main branch, and version 0.7.0 has been [published](https://crates.io/crates/dstv) on crates.io. Let me know if this solves your problem.